syncstorage-db-common/src/lib.rs

```diff
@@ -0,0 +1,304 @@
+pub mod error;
+pub mod params;
+pub mod results;
+pub mod util;
+
+use std::fmt::Debug;
+
+use async_trait::async_trait;
+use futures::{future, TryFutureExt};
+use lazy_static::lazy_static;
+use serde::Deserialize;
+use syncserver_db_common::{DbFuture, GetPoolState};
+
+use error::DbErrorIntrospect;
+use util::SyncTimestamp;
+
+lazy_static! {
+    /// For efficiency, it's possible to use fixed pre-determined IDs for
+    /// common collection names.  This is the canonical list of such
+    /// names.  Non-standard collections will be allocated IDs starting
+    /// from the highest ID in this collection.
+    pub static ref STD_COLLS: Vec<(i32, &'static str)> = {
+        vec![
+        (1, "clients"),
+        (2, "crypto"),
+        (3, "forms"),
+        (4, "history"),
+        (5, "keys"),
+        (6, "meta"),
+        (7, "bookmarks"),
+        (8, "prefs"),
+        (9, "tabs"),
+        (10, "passwords"),
+        (11, "addons"),
+        (12, "addresses"),
+        (13, "creditcards"),
+        ]
+    };
+}
+
+/// Rough guesstimate of the maximum reasonable life span of a batch
+pub const BATCH_LIFETIME: i64 = 2 * 60 * 60 * 1000; // 2 hours, in milliseconds
+
+/// The ttl to use for rows that are never supposed to expire (in seconds)
+pub const DEFAULT_BSO_TTL: u32 = 2_100_000_000;
+
+/// Non-standard collections will be allocated IDs beginning with this value
+pub const FIRST_CUSTOM_COLLECTION_ID: i32 = 101;
+
+#[async_trait]
+pub trait DbPoolTrait: Sync + Send + Debug + GetPoolState {
+    type Error;
```

Comment:
We need to introduce an associated type here because implementers of this trait have their own `DbError` types